CI: golangci-lint timeout when running with -j4 parallel flag

### What happened?

The make -j4 checks command in the GitHub Actions workflow times out when running golangci-lint, causing the CI pipeline to fail.

Error Output
```
level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
make: *** [checks.mk:44: lint] Error 4
Error: Process completed with exit code 2.
```

### How can we reproduce it?

In GitHub Actions (CI environment):
Push any commit to trigger the workflow
Observe the checks job in .github/workflows/tests.yml
The make -j4 checks step will timeout

### Environment

Location: .github/workflows/tests.yml
Command: make -j4 checks (line 30)
golangci-lint timeout: Currently set to 4m (--timeout=4m)

### Additional Information

_No response_

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -27,7 +27,7 @@ jobs:
           go-version-file: "go.mod"
           cache-dependency-path: "**/*.sum"
       - run: make -j2 install-tools install-linter-tool
-      - run: make -j4 checks
+      - run: make checks
 
   utest:
     needs:

